ansible/templates/minigraph_link_meta.j2

```diff
@@ -53,34 +53,23 @@
   </LinkMetadataDeclaration>
 {% endif %}
 
-{% if msft_an_enabled is defined and vm_topo_config.get('autoneg_interfaces') is not none %}
-  <LinkMetadataDeclaration>
+{% set num_of_intf = port_alias | length %}
+ <LinkMetadataDeclaration>
     <Link xmlns:a="http://schemas.datacontract.org/2004/07/Microsoft.Search.Autopilot.Evolution">
-{% for if_index in vm_topo_config['autoneg_interfaces']['intfs'] %}
-
-{% if "mellanox" in device_info[inventory_hostname]['HwSku']|lower %}
-{% set autoneg_intf = "etp" ~ if_index %}
-{% else %}
-{% set autoneg_intf = "Ethernet" ~ if_index ~ "/1" %}
-{% endif %}
-{% if device_conn[inventory_hostname][port_alias_map[autoneg_intf]]['autoneg']|lower == "on" %}
-        <a:LinkMetadata>
-            <a:Name i:nil="true"/>
-            <a:Properties>
-            <a:DeviceProperty>
-                 <a:Name>AutoNegotiation</a:Name>
-                 <a:Value>True</a:Value>
-            </a:DeviceProperty>
-            <a:DeviceProperty>
-                 <a:Name>FECDisabled</a:Name>
-                 <a:Reference i:nil="true"/>
-                 <a:Value>True</a:Value>
-            </a:DeviceProperty>
-            </a:Properties>
-            <a:Key>{{ device_conn[inventory_hostname][port_alias_map[autoneg_intf]]['peerdevice'] }}:{{ device_conn[inventory_hostname][port_alias_map[autoneg_intf]]['peerport'] }};{{ inventory_hostname }}:{{ autoneg_intf }}</a:Key>
-        </a:LinkMetadata>
-{% endif %}
+{% for index in range(num_of_intf) %}
+      {% if 'autoneg' in device_conn[inventory_hostname][port_alias_map[port_alias[index]]] %}
+      <a:LinkMetadata>
+        <a:Name i:nil="true"/>
+        <a:Properties>
+          <a:DeviceProperty>
+            <a:Name>AutoNegotiation</a:Name>
+            <a:Reference i:nil="true"/>
+            <a:Value>{{ device_conn[inventory_hostname][port_alias_map[port_alias[index]]]['autoneg'] }}</a:Value>
+          </a:DeviceProperty>
+        </a:Properties>
+        <a:Key>{{ inventory_hostname }}:{{ port_alias[index] }};{{ device_conn[inventory_hostname][port_alias_map[port_alias[index]]]['peerdevice'] }}:{{ device_conn[inventory_hostname][port_alias_map[port_alias[index]]]['peerport'] }}</a:Key>
```

Comment:
Will all naming conventions work here ?
So Vendors use etp1, etp2 etc as port alais

Reply:
Removed relation to port alias, what will be updated in link.csv will be taken here